portal/src/main/webapp/js/src/study-view/component/PieChart.js

```diff
@@ -960,21 +967,22 @@ var PieChart = function(){
 
     //Called when the number of label biggen than 6, used by addPieLabels()
     function labelFunction() {
-        var _tableDiv = '';
+        var _tableDiv = [];
+        var _tableDivStr = '';
         var _labelLT = labelWLT; //label length threshold
         var _labelHeaderLT = labelHeaderWLT; //label header length threshold
 
-        _tableDiv += '<table';
+        _tableDiv.push('<table');
 
         if(category[0] === 'w1') {
             _labelLT = labelLT;
             _labelHeaderLT = labelHeaderLT;
-            _tableDiv += ' class="smallTable" ';
+            _tableDiv.push(' class="smallTable" ');
         }else{
-            _tableDiv += ' class="regularTable" ';
+            _tableDiv.push(' class="regularTable" ');
         }
 
-        _tableDiv += '><thead><th>'+selectedAttrDisplay+'</th><th>#</th><th>Freq</th></thead><tbody>';
+        _tableDiv.push('><thead><th>'+selectedAttrDisplay+'</th><th>#</th><th>Freq</th></thead><tbody>');
```

Comment:
so, based on comment above, this could be done with something like: 

```javascript
table.append(
    $('<thead/>').append(
        $('<th/>', {text: selectedAttrDisplay})
    )
    .append(
        $('<th/>', {text: '#'})
    )
    .append(
        $('<th/>', {text: 'Freq'})
    );
var tableBody = table.append('<tbody/>');
```